context/shared-icons.tsx

```diff
@@ -0,0 +1,37 @@
+'use client';
+
+import { useState } from 'react';
+import { useContext, createContext } from 'react';
+import { HiddenReusableIcon, AvailableIcons } from '@/components/common/Icon';
+
+interface SharedIcons {
+  iconReferences: Array<string>;
+  setIconReferences: React.Dispatch<React.SetStateAction<Any>>;
+}
+
+const SharedIconContext = createContext<SharedIccons | undefined>(undefined);
+
+export function SharedIconProvider(props: React.PropsWithChildren) {
+  //const [iconReferences, setIconReferences] = useState([]);
+  const iconReferences = Object.values(AvailableIcons);
+  const hiddenIcons = iconReferences.map((iconRef) => (
+    <HiddenReusableIcon key={iconRef} name={iconRef} />
+  ));
+
+  return (
+    <SharedIconContext.Provider value={{ iconReferences }}>
+      <div style={{ display: 'none' }}>{hiddenIcons}</div>
+      {props.children}
+    </SharedIconContext.Provider>
```

Comment:
Is context necessary here? Would it be enough to just render the `hiddenIcons`

Reply:
No, there was an earlier, too complicated version where the context was actually updated based on which icons are actually used, but I think that's unnecessary. Will simplify.